fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -359,6 +363,26 @@ public void updateUniquenessViolations(@Nonnull Tuple valueKey, @Nonnull Tuple p
         }
     }
 
+    @Nonnull
+    @Override
+    public CompletableFuture<Void> removeUniquenessViolationsAsync(@Nonnull Tuple valueKey, @Nonnull Tuple primaryKey) {
+        Subspace uniqueValueSubspace = state.store.indexUniquenessViolationsSubspace(state.index).subspace(valueKey);
+        state.transaction.clear(uniqueValueSubspace.pack(primaryKey));
+        // Remove the last entry if it was the second last entry in the unique value subspace.
+        RecordCursor<KeyValue> uniquenessViolationEntries = KeyValueCursor.Builder.withSubspace(uniqueValueSubspace)
+                .setContext(state.context)
+                .setScanProperties(new ScanProperties(ExecuteProperties.newBuilder()
+                        .setReturnedRowLimit(2)
+                        .setIsolationLevel(IsolationLevel.SERIALIZABLE)
+                        .build()))
```

Comment:
I think we probably also want to set the streaming mode on this scan properties to `CursorStreamingMode.WANT_ALL`